### ISSUE TITLE
no longer functions on firefox

### ISSUE THREAD

**Issue by 5paceToast** [asciidoctor/asciidoctor-browser-extension#351]
Trying to load a raw github url: https://raw.githubusercontent.com/asciidoctor/asciidoctor-browser-extension/master/README.adoc - nothing happens.
Similarly, a raw file:// url also does not load: file:///home/toast/Workspace/toast/libuconf/README.adoc

This started happening a while ago for me, on all my browsers (firefox), I only got around to reporting it right now.
Current version is 2.4.0 for me.

**Comment by Mogztter** [comment]
Hello @5paceToast 

Sorry about that, I had to publish an "emergency" version for Chrome to fetch the content from the background script because starting with version 80, `XMLHttpRequest` sent by the content script are blocked.
Unfortunately, this solution does not work on Firefox, see: https://discourse.mozilla.org/t/get-the-raw-content-of-the-current-page-mission-impossible/54807

So until this issue is fixed, I have no choice but to publish one version for Firefox and one version for Chrome... :disappointed: 

Long story short, I've unpublished version 2.4.0 from the Firefox Add-on weeks ago. So if you "update" the add-on you should downgrade to version 2.3.1: https://github.com/asciidoctor/asciidoctor-browser-extension/releases/tag/v2.3.1

**Comment by 5paceToast** [comment]
@Mogztter is there an easy way to force updates on firefox?
I have it set to auto-update, and it doesn't seem to downgrade.
I don't want to uninstall and reinstall because I'd have to go tweak stuff on all my browsers afterwards (since I have addons sync), which is like 5-6 different ones.

**Comment by 5paceToast** [comment]
Also, maybe this should be documented somewhere easily visible, in case more people run into this problem? ^^
Perhaps in the same warning section on the README.

**Comment by Mogztter** [comment]
> is there an easy way to force updates on firefox?
I have it set to auto-update, and it doesn't seem to downgrade.
I don't want to uninstall and reinstall because I'd have to go tweak stuff on all my browsers afterwards (since I have addons sync), which is like 5-6 different ones.

I thought that Firefox would automatically downgrade to the latest version published... I'm not 100% sure but I think that what Firefox did on my machine because I'm now on version 2.3.1.

> Also, maybe this should be documented somewhere easily visible, in case more people run into this problem? ^^
Perhaps in the same warning section on the README.

If Firefox does not automatically revert to the latest version published then yes we should probably add an information message to tell users to manually downgrade to 2.3.1. But I cannot believe that this is not a built-in feature of the Firefox Addon.
Did you manually install the extension in the first place (using the xpi)?

@jwflory @nigjo @r-pf Did you reinstall the extension or did Firefox automatically downgrade to 2.3.1?

**Comment by 5paceToast** [comment]
> Did you manually install the extension in the first place (using the xpi)?
No, I installed it from addons.mozilla.org

When I pressed "check for updates" it told me "no updates found".
I guess the firefox update checker compares against the local version, in case you're doing local dev (?)

**Comment by jwflory** [comment]
> @jwflory @nigjo @r-pf Did you reinstall the extension or did Firefox automatically downgrade to 2.3.1?

I removed the v2.4.0 add-on to manually reinstall v2.3.1.

**Comment by nigjo** [comment]
nearly the same here. Removed 2.4 to reinstall 2.3.1 from Mozilla

**Comment by Mogztter** [comment]
So it's not automatic... that's sad :disappointed: 

> Also, maybe this should be documented somewhere easily visible, in case more people run into this problem? ^^
Perhaps in the same warning section on the README.

We should definitely do that then!
So the procedure is to manually remove version 2.4.0 and then reinstall 2.3.1 from the Mozilla Add-ons website (or from the GitHub release page).

**Comment by Mogztter** [comment]
Actually I think I can publish a version 2.4.1 with the same content as 2.3.1. So when a user will try to upgrade it will remove 2.4.0 and install 2.4.1. 
I've asked the AMO to make sure that this is the correct way to do it: https://discourse.mozilla.org/t/automatically-downgrade-users-to-the-latest-version-published-enabled/58359 but it will avoid a lot of uninstall/reinstall shenanigans

**Comment by jwflory** [comment]
> Actually I think I can publish a version 2.4.1 with the same content as 2.3.1. So when a user will try to upgrade it will remove 2.4.0 and install 2.4.1.

This makes the most sense to me! :+1:

**Comment by Mogztter** [comment]
Version 2.4.1 has been published and should be available in the next few days (pending review)

https://github.com/asciidoctor/asciidoctor-browser-extension/releases/tag/firefox-2.4.1

**Comment by Mogztter** [comment]
Version 2.4.1 is live! https://github.com/asciidoctor/asciidoctor-browser-extension/issues/356#issuecomment-624801137

Closing.... sorry about the disruption :sweat_smile:
